### PR TITLE
fix: don't panic in redis library on creating connection pool

### PR DESCRIPTION
**Pull request by dracarys18**
## Type of Change

- [x] Enhancement

## Description

Don't panic while creating connection pool in redis library.

## Motivation and Context

Removing panics in library code

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code

### CODE REVIEW

File: crates/redis_interface/src/lib.rs [hunk 1]
```diff
@@ -21,24 +21,20 @@ pub mod commands;
...
 
 impl RedisConnectionPool {
     /// Create a new Redis connection
-    ///
-    /// # Panics
-    ///
-    /// Panics if a connection to Redis is not successful.
-    #[allow(clippy::expect_used)]
-    pub async fn new(conf: &RedisSettings) -> Self {
+    pub async fn new(conf: &RedisSettings) -> Result<Self, RedisError> {
```
Comment (SanchithHegde) [inline crates/redis_interface/src/lib.rs:37]: Do you think we should raise our error type instead, an `errors::RedisError::RedisConnectionError` variant maybe?

**Review by SanchithHegde — COMMENTED** [review]
Other than that, looks good to me!

File: crates/redis_interface/src/lib.rs (hunk 1 above)
Comment (dracarys18) [reply crates/redis_interface/src/lib.rs:37]: i guess we could something like 
```rust
RedisConnectionError {err: fred::RedisError}
```

Comment (SanchithHegde) [reply crates/redis_interface/src/lib.rs:37]: Either that, or return an `error_stack::Result` / `CustomResult`.

**Review by SanchithHegde — APPROVED** [review]

**Review by ashokkjag — APPROVED** [review]

**Pull request merged**